x-pack/platform/plugins/shared/ingest_pipelines/public/application/components/pipeline_editor/components/processor_form/processors/remove.tsx

```diff
@@ -5,49 +5,156 @@
  * 2.0.
  */
 
-import React, { FunctionComponent } from 'react';
+import React, { FunctionComponent, useCallback, useMemo, useState } from 'react';
 import { i18n } from '@kbn/i18n';
 
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiLink, EuiText } from '@elastic/eui';
+import { isEmpty } from 'lodash';
 import {
   FIELD_TYPES,
   UseField,
   ComboBoxField,
-  fieldValidators,
+  FieldHook,
+  FieldConfig,
+  useFormContext,
 } from '../../../../../../shared_imports';
 
-import { FieldsConfig, to } from './shared';
+import { to } from './shared';
 
 import { IgnoreMissingField } from './common_fields/ignore_missing_field';
 
-const { emptyField } = fieldValidators;
-
-const fieldsConfig: FieldsConfig = {
-  field: {
-    type: FIELD_TYPES.COMBO_BOX,
-    deserializer: to.arrayOfStrings,
-    serializer: (v: string[]) => (v.length === 1 ? v[0] : v),
-    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.removeForm.fieldNameField', {
-      defaultMessage: 'Fields',
-    }),
-    helpText: i18n.translate('xpack.ingestPipelines.pipelineEditor.removeForm.fieldNameHelpText', {
-      defaultMessage: 'Fields to remove.',
-    }),
-    validations: [
-      {
-        validator: emptyField(
-          i18n.translate('xpack.ingestPipelines.pipelineEditor.removeForm.fieldNameRequiredError', {
-            defaultMessage: 'A value is required.',
-          })
-        ),
-      },
-    ],
-  },
+interface RemoveFieldsTypes {
+  fields_to_remove: string[];
+  fields_to_keep: string[];
+}
+
+type RemoveFields = {
+  [K in keyof RemoveFieldsTypes]: FieldHook<RemoveFieldsTypes[K]>;
 };
 
+const getFieldConfig = (
+  toggleField: () => void
+): Record<
+  keyof RemoveFields,
+  {
+    path: string;
+    config: FieldConfig<any>;
+    labelAppend: JSX.Element;
+    key: string;
+  }
+> => ({
+  fields_to_remove: {
+    path: 'fields.field',
+    config: {
+      type: FIELD_TYPES.COMBO_BOX,
+      deserializer: to.arrayOfStrings,
+      serializer: (v: string[]) => (v.length === 1 ? v[0] : v),
+      fieldsToValidateOnChange: ['fields.field', 'fields.keep'],
+      label: i18n.translate('xpack.ingestPipelines.pipelineEditor.removeForm.fieldNameField', {
+        defaultMessage: 'Fields to remove',
+      }),
+      helpText: i18n.translate(
+        'xpack.ingestPipelines.pipelineEditor.removeForm.fieldNameHelpText',
+        {
+          defaultMessage: 'You can use plain text or template snippets.',
```

Comment:
If we're not going to show an example of what a template snippet looks like, we could link to the docs